cylc/flow/scripts/lint.py

```diff
@@ -88,156 +223,175 @@
+                \s*
+                # none, None and root are valid family names
+                # and `inherit =` or `inherit = # x` are valid too
+                (['"]?(none|None|root|\#.*|$)['"]?|
+                (
+                    # as are families named with capital letters
+                    [A-Z0-9_-]+
+                    # and optional quotes
+                    | [\'\"]
+                    # which may include Cylc parameters
+                    | (<[^>]+>)
+                    # or Jinja2
+                    | ({[{%].*[%}]})
+                    # or EmPy
+                    | (@[\[{\(]).*([\]\}\)])
+                )+
+                )
+                # this can be a comma separated list
+                (
+                \s*,\s*
+                # none, None and root are valid family names
+                (['"]?(none|None|root)['"]?|
+                    (
+                    # as are families named with capital letters
+                    [A-Z0-9_-]+
+                    # and optional quotes
+                    | [\'\"]
+                    # which may include Cylc parameters
+                    | (<[^>]+>)
+                    # or Jinja2
+                    | ({[{%].*[%}]})
+                    # or EmPy
+                    | (@[\[{\(]).*([\]\}\)])
+                    )+
+                )
+                )*
+                # allow trailing commas and whitespace
+                \s*,?\s*
+                # allow trailing comments
+                (\#.*)?
+                $
+            )
+            ''',
+            re.X
+        ).findall,
     },
-    re.compile(r'{[{%]'): {
+    "S008": {
         'short': JINJA2_FOUND_WITHOUT_SHEBANG,
         'url': '',
-        'index': 8
+        'kwargs': True,
+        FUNCTION: check_jinja2_no_shebang,
+        'fallback': re.compile(r'{[{%]').findall
+        # FUNCTION: re.compile(r'{[{%]').findall,
     },
-    re.compile(r'platform\s*=\s*\$\(.*?\)'): {
+    "S009": {
         'short': 'Host Selection Script may be redundant with platform',
         'url': (
             'https://cylc.github.io/cylc-doc/stable/html/7-to-8/'
             'major-changes/platforms.html'
         ),
-        'index': 9
+        FUNCTION: re.compile(r'platform\s*=\s*\$\(.*?\)').findall,
     },
-    re.compile(r'platform\s*=\s*(`.*?`)'): {
+    "S010": {
         'short': 'Using backticks to invoke subshell is deprecated',
         'url': 'https://github.com/cylc/cylc-flow/issues/3825',
-        'index': 10
+        FUNCTION: re.compile(r'platform\s*=\s*(`.*?`)').findall,
     },
-    re.compile(r'(?<!{)#.*?{[{%]'): {
-        'short': 'Cylc comments (#) do not prevent Jinja2 processing.',
+    "S011": {
+        'short': 'Cylc will process commented Jinja2!',
         'url': '',
-        'index': 11
+        'kwargs': True,
+        'evaluate commented lines': True,
+        FUNCTION: check_if_jinja2,
+        'fallback': re.compile(r'(?<!{)#.*?{[{%]').findall,
```

Comment:
Instead of this "fallback" approach could you not use `functools.partial`?
```suggestion
        FUNCTION: partial(
            check_if_jinja2,
            pattern=re.compile(r'(?<!{)#.*?{[{%]')
        ),
```